### PR TITLE
[Fix] Fix ci

### PR DESCRIPTION
**Pull request by HAOCHENYE**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

- docformatter: https://github.com/PyCQA/docformatter/issues/289.
- flake8:  https://stackoverflow.com/questions/77401175/how-to-make-flake8-ignore-syntax-within-strings
- pre-commit-hooks: https://github.com/pre-commit/pre-commit-hooks/issues/971
- github-action: https://github.blog/changelog/2024-03-07-github-actions-all-actions-will-run-on-node20-instead-of-node16-by-default/

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

**Before PR**:

- [ ] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) to create this PR.
- [ ] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) are used to fix the potential lint issues.
- [ ] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with some of those projects, like MMDet or MMCls.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: .pre-commit-config-zh-cn.yaml [hunk 1]
```diff
@@ -13,7 +17,7 @@ repos:
     hooks:
       - id: yapf
   - repo: https://gitee.com/openmmlab/mirrors-pre-commit-hooks
-    rev: v4.3.0
+    rev: v5.0.0
```
Comment (zhouzaida) [inline .pre-commit-config-zh-cn.yaml:20]: 没有这个分支

File: .pre-commit-config-zh-cn.yaml [hunk 2]
```diff
@@ -53,15 +58,16 @@ repos:
...
+  - repo: https://github.com/pre-commit/mirrors-mypy
+    rev: v1.2.0
     hooks:
       - id: mypy
         exclude: |-
           (?x)(
               ^test
               | ^docs
           )
+        additional_dependencies: ["types-setuptools", "types-requests"]
```
Comment (zhouzaida) [inline .pre-commit-config-zh-cn.yaml:70]: 和 mmengine 的依赖不一致，https://github.com/open-mmlab/mmengine/pull/1598/files#diff-63a9c44a44acf85fea213a857769990937107cf072831e1a26808cfde9d096b9R66

File: mmcv/ops/deform_conv.py [hunk 3]
```diff
@@ -449,7 +449,9 @@ def _load_from_state_dict(self, state_dict, prefix, local_metadata, strict,
 
     @MODELS.register_module('DCN', force=True)
     class DeformConv2dPack_MLU(DeformConv2d):
-        """This class is the DCN implementation of the MLU device. The MLU
+        """This class is the DCN implementation of the MLU device.
+
+        The MLU
```
Comment (zhouzaida) [inline mmcv/ops/deform_conv.py:454]: 这一行有点奇怪，应该和下面的合在一起？

**Comment by zhouzaida** [comment]
python 脚本我都看了，没啥问题，但 cpp cu 的格式风格可能要再 check 一下 clang 的版本

File: .pre-commit-config-zh-cn.yaml (hunk 1 above)
Comment (HAOCHENYE) [reply .pre-commit-config-zh-cn.yaml:20]: 后续有需求提 issue 的话 gitte 里同步吧。我理解现在应该几乎木有人用 zh-cn 的版本

**Pull request merged**